src/bootsect/bootsect.asm

```diff
@@ -14,7 +17,25 @@ start:
     mov ss, ax
     mov sp, 0x7c00
     sti
-
+    
+    ; Some BIOSes don't pass the correct boot drive number,
+    ; so we need to do the job
+  .check_drive:
+    ; Qloader2 isn't made for floppy disks, these are dead anyways.
+    ; So if the value the BIOS passed is <0x80, just assume it has passed
+    ; an incorrect value
+    test dl, 0x80
+    jz .fix_drive
+
+    ; Drive numbers from 0x80..0x8f should be valid
+    test dl, 0x70
+    jz .continue
+
+  .fix_drive:
+    ; Try to fix up the mess the BIOS have done
+    mov dl, 0x80
```

Comment:
Could be worth reading back up some flag to make sure the new drive is not a bogus one and gracefully handle that scenario?

Reply:
sorry, I didn't understand exactly what you mean. Are you saying about testing if the new drive is valid.

Reply:
What I meant to say is whether we should check for a signature after loading the other sectors in, other than just checking for a successful call to int 13h, to make sure we are reading from the correct drive (0x80) and not some other unrelated drive/bogus data.